### PR TITLE
Prototype of a setup.py

### PR DESCRIPTION
**Pull request by ocefpaf**
@miguelcleon this is not ready to be merged. It is just a prototype to start a discussion. See my inline comments.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -2,14 +2,21 @@ language: python
...
+    env: TEST_TARGET=default
+  - python: 3.4
+    env: TEST_TARGET=default
+  - python: 3.5
+    env: TEST_TARGET=default
+  - python: 3.5
+    env: TEST_TARGET=coding_standards
+  allow_failures:
+  - python: 3.5
+    env: TEST_TARGET=coding_standards
```
Comment (ocefpaf) [inline .travis.yml:18]: This adds a better test matrix that reports but allows coding standards failures. See https://travis-ci.org/miguelcleon/ODM2-Admin/builds/164040616

File: setup.py [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+        TestCommand.finalize_options(self)
+        self.verbose = True
+
+    def run_tests(self):
+        import pytest
+        errno = pytest.main(self.test_args)
+        sys.exit(errno)
+
+
+def extract_version(module='odm2djangoadmin'):
```
Comment (ocefpaf) [inline setup.py:18]: We need to decide the main `import` name for this module. I used `odm2djangoadmin` in the previous PR, is that still OK?

File: setup.py [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+    version = None
+    fdir = os.path.dirname(__file__)
+    fnme = os.path.join(fdir, module, '__init__.py')
```
Comment (ocefpaf) [inline setup.py:21]: This pattern allows to define the module version only in the `__init__.py` and read it here later. Some people define it twice and it is easy diverge and get inconsistent info about the version.

File: setup.py [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+    return version
+
+rootpath = os.path.abspath(os.path.dirname(__file__))
+
+
+def read(*parts):
+    return open(os.path.join(rootpath, *parts), 'r').read()
+
+
+long_description = '{}'.format(read('README.rst'))
```
Comment (ocefpaf) [inline setup.py:38]: I recommend we switch the README syntax from Markdown to ReST. GitHub renders both but PyPI renders only the latter. (I have another PR with that change ready to go.)

File: setup.py [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+      classifiers=['Development Status :: 5 - Production/Stable',
+                   'Environment :: Console',
+                   'Intended Audience :: Science/Research',
+                   'Intended Audience :: Developers',
+                   'Intended Audience :: Education',
+                   'License :: OSI Approved :: MIT License',
+                   'Operating System :: OS Independent',
+                   'Programming Language :: Python',
+                   'Topic :: Scientific/Engineering',
+                   'Topic :: Education',
```
Comment (ocefpaf) [inline setup.py:58]: Please review the classifier info and let me know if you request any changes.

File: setup.py [hunk 6]
```diff
@@ -0,0 +1,75 @@
...
+                   ],
+      description='Django admin app for Observation Data Model 2 (ODM2)',
+      url='https://github.com/miguelcleon/odm2djangoadmin',
```
Comment (ocefpaf) [inline setup.py]: Note to self: fix the URL.

**Review by ocefpaf — COMMENTED** [review]
Added a few inline comments.

File: setup.py (hunk 2 above)
Comment (miguelcleon) [reply setup.py:18]: it should be ODM2-Admin, or odm2-admin if it needs to be lower case, or odm2admin if we can't use a '-'

File: setup.py (hunk 4 above)
Comment (miguelcleon) [reply setup.py:38]: ok

**Pull request merged**